src/sdam/server.ts

```diff
@@ -148,9 +148,10 @@ export class Server extends TypedEventEmitter<ServerEvents> {
       options,
       state: STATE_CLOSED,
       topology,
-      pool: new ConnectionPool(this, poolOptions),
       operationCount: 0
-    };
+    } as any;
+    // NOTE: this is done so that the connection pool can access the topology in its constructor
```

Comment:
I don't see anything in the constructor of the connection pool that would require the topology present on `server.s`.

How come this change is necessary?

Reply:
The `ConnectionPool's` `client` getter accesses the `topology` field of `server.s` and since we have to make a call to the logger once the `ConnectionPool`'s constructor exits (in the `process.nextTick` callback), we were getting a `TypeError` for accessing the `client` field on the undefined `topology`

Reply:
Can we consider a restructure? I know we have a ticket somewhere but we've wanted to migrate away from the `s` subobject pattern where we can, it doesn't play well with TS or I guess runtime contracts. The Server class should be able to advertise an always assigned Topology property as soon as it exists. I'm thinking we store it as a top-level class property

Reply:
Should we also do the same with the connection pool?